src/datapool/datapool_pmem.c

```diff
@@ -31,11 +34,16 @@
  * opened.
  */
 struct datapool_header {
-    uint8_t signature[DATAPOOL_SIGNATURE_LEN];
-    uint64_t version;
-    uint64_t size;
-    uint64_t flags;
-    uint8_t unused[DATAPOOL_HEADER_LEN - 32];
+    struct {
```

Comment:
it's not really clear to me why these need to be in their own structs, rather than adding a user_data member

Reply:
And it should be the other way around. The signature and version should be first thing in the file. If only so that magic file isn't excessively complicated.

Now that I think about it, we should also add `user_signature` or a `layout` field, so that when there are two pelikan database files we can programmatically distinguish between them, and refuse to open, e.g., cuckoo hashtable in the slab engine, avoiding a segfault.

Reply:
I am working on it.

Reply:
Done. User signature field was added.